app/utils/UserAnswers.scala

```diff
@@ -114,12 +115,16 @@ case class UserAnswers(json: JsValue = Json.obj()) {
   }
 
   def set[I <: TypedIdentifier.PathDependent](id: I)(value: id.Data)(implicit writes: Writes[id.Data]): JsResult[UserAnswers] = {
-
     val jsValue = Json.toJson(value)
-
-    JsLens.fromPath(id.path)
-      .set(jsValue, json)
-      .flatMap(json => id.cleanup(Some(value), UserAnswers(json)))
+    val oldValue = json
+    val jsResultSetValue = JsLens.fromPath(id.path).set(jsValue, json)
+    jsResultSetValue.flatMap { newValue =>
+      if (oldValue == newValue) {
+        JsSuccess(UserAnswers(newValue))
+      } else {
+        jsResultSetValue.flatMap(json => id.cleanup(Some(value), UserAnswers(json)))
+      }
```

Comment:
The code bit 'jsResultSetValue.flatMap(json => ' is the same as the 'newValue' which is already available and is not needed again

Reply:
Fixed